tools/fds/server.cpp

```diff
@@ -188,16 +207,20 @@ int main (
     }
     else
     {
-        // Wait for user command
-        cout << "\n### Server is running, press any key to quit ###" << std::endl;
-        cout.flush();
-        cin.ignore();
+        unique_lock<mutex> lock(g_signal_mutex);
+
+        // handle signal SIGINT for every thread
+        signal(SIGINT, sigint_handler);
+        cout << "\n### Server is running ###" << std::endl;
+
+        g_signal_cv.wait(lock, []
+                {
+                    return 0 != g_signal_status;
+                });
 
-        // Remove the server
-        Domain::removeParticipant(pServer);
```

Comment:
Shouldn't we remove the Participant for a clean exit, anyways?

Reply:
The later called to `Domain::stopAll()` takes care of it.